packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2800,6 +2808,10 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   void didChangeDependencies() {
     super.didChangeDependencies();
 
+    _scrollNotificationObserver?.removeListener(_handleContextMenuOnParentScroll);
```

Comment:
Every editable field in the app now needs to add itself as a listener. Can the logic be moved to the context menu or something (I think there was a singleton class since there can be at most one menu visible), or only add itself as the listener when the menu becomes visible?

Reply:
Also my understanding is this notifies listeners when **any** descendent scrolls, even when the scrolling scrollable and the active text field are in different subtrees. That probably should cause the toolbar to disappear as it is not scrolling at all.

Reply:
The documentation states that: 

> Notifications bubble up through the tree, which means a given [NotificationListener](https://main-api.flutter.dev/flutter/widgets/NotificationListener-class.html) will receive notifications for all descendant [Scrollable](https://main-api.flutter.dev/flutter/widgets/Scrollable-class.html) widgets. To focus on notifications from the nearest [Scrollable](https://main-api.flutter.dev/flutter/widgets/Scrollable-class.html) descendant, check that the [depth](https://main-api.flutter.dev/flutter/widgets/ViewportNotificationMixin/depth.html) property of the notification is zero.

So I think that means only `Scrollable`s send notifications so you can do the more efficiently by going up the `Scrollable` tree instead (but you may want to avoid calling `Scrollable.of` since it establishes a dependency).

Reply:
Yeah, in this case, you would likely not want to check notification depth and limit it to 0. The depth increases for each nested scrollable.